app/src/main/java/fr/free/nrw/commons/category/CategoryEditHelper.java

```diff
@@ -62,16 +62,27 @@ private Observable<Boolean> addCategory(Media media, List<String> categories,
         final String wikiText) {
         Timber.d("thread is category adding %s", Thread.currentThread().getName());
         String summary = "Adding categories";
-
         final StringBuilder buffer = new StringBuilder();
-
-        final String wikiTextWithoutCategory
-            = wikiText.substring(0, wikiText.indexOf("[[Category"));
-
-        if (categories != null && categories.size() != 0) {
+        final String wikiTextWithoutCategory;
+        //If the picture update with skipping the category setting, the wikiText will contain "Uncategorized" instead of "[[Category"
+        if (wikiText.contains("Uncategorized")) {
+            wikiTextWithoutCategory = wikiText.substring(0, wikiText.indexOf("Uncategorized"));
+        } else if (wikiText.contains("[[Category")) {
+            wikiTextWithoutCategory = wikiText.substring(0, wikiText.indexOf("[[Category"));
+        } else {
+            wikiTextWithoutCategory = "";
+        }
+        if (categories != null && !categories.isEmpty()) {
+           //If categories list is empty,when reading category for picture,
```

Comment:
-> `If the categories list is empty, when reading the categories of a picture,

Reply:
Thanks, I update the comments